server/src/main/java/org/candlepin/controller/Entitler.java

```diff
@@ -191,21 +199,35 @@ public void adjustEntitlementQuantity(Consumer consumer, Entitlement ent,
      * @return List of Entitlements
      * @throws AutobindDisabledForOwnerException when an autobind attempt is made and the owner
      *         has it disabled.
+     * @throws AutobindHypervisorDisabledException when an autobind attempt is made on a hypervisor
+     *         and the owner has it disabled.
      */
     @Transactional
     public List<Entitlement> bindByProducts(AutobindData data, boolean force)
-        throws AutobindDisabledForOwnerException {
+        throws AutobindDisabledForOwnerException, AutobindHypervisorDisabledException {
         Consumer consumer = data.getConsumer();
         Owner owner = data.getOwner();
-
-        if ((!consumer.isDev() && owner.isAutobindDisabled()) || owner.isContentAccessEnabled()) {
+        ConsumerType type = this.consumerTypeCurator.getConsumerType(consumer);
+        if (!consumer.isDev() && (owner.isAutobindDisabled() ||
```

Comment:
Is there any way we can pair down this beast of a conditional? Seems like we're doing a lot of very-specific context checking for this code.

If not, and if it's worth doing, we could add a simple utility function to ConsumerTypeEnum that could do the inverse check safely without needing to pretest for null:

```
public boolean isType(ConsumerType type) {
  return type != null && type.isType(this); // Alternatively, extend this out to not call into the type's isType method
}
```

Reply:
I made it easier to read. Good enough?

Reply:
Yeah, that's much better. Thanks